src/plugins/visualize/public/application/utils/use/use_visualize_app_state.tsx

```diff
@@ -90,10 +90,20 @@ export const useVisualizeAppState = (
       // The savedVis is pulled from elasticsearch, but the appState is pulled from the url, with the
       // defaults applied. If the url was from a previous session which included modifications to the
       // appState then they won't be equal.
-      if (!isEqual(stateContainer.getState().vis, stateDefaults.vis)) {
+      if (
+        !isEqual(stateContainer.getState().vis, stateDefaults.vis) ||
```

Comment:
are you sure we need to execute `stateContainer.getState()` 6 times during executing one sync method?

Reply:
done